### PR TITLE
gensio: Add new package

### PR DESCRIPTION
**Pull request by WereCatf**
A library to abstract stream I/O like serial port, TCP, telnet,
UDP, SSL, IPMI SOL, etc.

Signed-off-by: Nita Vesa <werecatf@outlook.com>

Maintainer: me
Compile tested: ath79, mvebu, ramips, lantiq with OpenWRT 21.02.2 SDK.
Run tested: Same as above, compiled new ser2net against gensio and tested that ser2net works.

### CODE REVIEW

**Comment by mhei** [comment]
This library has been on my todo list for a long time, great to see this PR.
From a first very quick point of view, this looks good to me. However, one question: the musl compat patch, did you discuss with upstream? Would be nice to have it included upstream, to not carry it here...

**Comment by WereCatf** [comment]
> This library has been on my todo list for a long time, great to see this PR. From a first very quick point of view, this looks good to me. However, one question: the musl compat patch, did you discuss with upstream? Would be nice to have it included upstream, to not carry it here...

No, I did not. Didn't actually occur to me, since this is such a tiny patch and I was just solely focused on getting this thing packaged up.

**Comment by WereCatf** [comment]
@mhei If you feel it'd be best to wait and see, if the patch could be upstreamed first, just say the word. Alas, I'm not well versed in cross-platform stuff and I have no idea, if the patch would then break something on some other platform and thus cause the developer to reject it.

I've got an updated version of ser2net waiting to see, if this gets approved or not, but I'm not in a hurry.

**Comment by mhei** [comment]
I think the patch is correct and should be applied upstream, although I'm no expert on cross-platform stuff either - let alone Windows :smile: But this should not delay this PR here, since we don't know when upstream creates next release. But you should at least try it and see, how it goes.

**Comment by WereCatf** [comment]
Jesus Christ this CI-thing just won't accept the patch. "Dirty patches detected" -- how do I make it "non-dirty?" Why does it even matter, when the patch itself applies without issue and compiles just fine?

**Comment by WereCatf** [comment]
One more try at getting the CI to accept the patch and if it still complains, I'm gonna blow a gasket.

**Comment by champtar** [comment]
Maybe you found but for dirty patch you can use `make package/geniso/refresh`

**Comment by WereCatf** [comment]
Well, I don't know how to fix the compilation errors. I guess it needs some cmake-magic, but I know nothing about any of that stuff.

**Comment by WereCatf** [comment]
Can one somehow restart the CI-build? It seems it failed at strongswan, not with my changes -- it never even got that far.

File: net/gensio/Config.in [hunk 1]
```diff
@@ -0,0 +1,39 @@
+	config GENSIO_SSL
+		depends on PACKAGE_libgensio
+		bool "Enable SSL-support via OpenSSL"
+		default n
+
+	config GENSIO_SSHD
+		depends on GENSIO_SSL
+		bool "Build gtlsshd"
```
Comment (mhei) [inline net/gensio/Config.in:8]: This should not be configurable in the menu of the library IMO. I'd prefer to have this as option of gensio-bin and then it should select GENSIO_SSL and not depend on it.

Comment (WereCatf) [reply net/gensio/Config.in]: Agreed, changed.

**Review by mhei — APPROVED** [review]

**Comment by mhei** [comment]
LGTM, thanks for your patience.

**Comment by hnyman** [comment]
Like mentioned in #18166, 
this Makefile contains a circular dependency:

```
feeds/packages/net/gensio/Config.in:11:error: recursive dependency detected!
feeds/packages/net/gensio/Config.in:11:	symbol GENSIO_SCTP depends on PACKAGE_libgensio
tmp/.config-package.in:53955:	symbol PACKAGE_libgensio depends on GENSIO_SCTP
For a resolution refer to Documentation/kbuild/kconfig-language.rst
subsection "Kconfig recursive dependency limitations"
```

cc @mhei @WereCatf

**Comment by WereCatf** [comment]
@hnyman I do not know how to fix that. It seems to only happen with the GENSIO_SCTP, none of the other configuration-options, even though there is literally zero difference between them. Renaming it to something else doesn't work and, as far as I know, it has to depend on PACKAGE_libgensio or else the configuration-options won't show up in the correct place.

Given how none of the other options trigger this error, it seems to be a bug with the build-system instead of gensio or there is something happening behind the scenes that I am unaware of.

I will happily send a patch, if someone explains to me how to fix this and what is actually causing the issue, so I know to be aware of it in the future.

**Pull request merged**